packages/store/src/operators/of-action.ts

```diff
@@ -8,7 +8,7 @@ export interface ActionCompletion<T = any> {
   result: {
     successful: boolean;
     canceled: boolean;
-    error?: Error;
+    error?: any;
```

Comment:
why any?

Reply:
can we somehow learn how to use the type we need

Reply:
I don't think so, because you can also have ReferenceError when you expect to something else.

Reply:
```
export interface ActionContext<T = any, U extends Error> {
  status: ActionStatus;
  action: T;
  error?: U;
}
```

Reply:
Error is an interface, so I guess that supposed to be `U implements Error`.
It might get messy if someone, for some reason, except multiple errors in the same action.

@markwhitfeld What is your preferred way?

Reply:
@eranshmil Is the problem with the Error type that a user may throw a string as an error instead?
(which doesn't extend error)

Reply:
That might be as a result of a runtime error (such as ReferenceError), a custom thrown error, HttpErrorResponse. (And I guess that there's more use cases)

Reply:
@eranshmil Those usually extend `Error` but I see your point about the casting, and we also cannot guarantee that what was thrown does extend `Error`.
How about something like this:
```TS

export interface ActionCompletion<T = any, TError = Error> {
  action: T;
  result: {
    successful: boolean;
    canceled: boolean;
    error?: TError;
  };
}
```
PS I called it `TError` because it is a bit more descriptive than `U`. Up to you though.

Reply:
Usually I'd agree with you about type safety, but it might be problematic if an unexpected error has happened.
Does it necessary mean that the error would implement the Error interface?

Reply:
No, in my sample it is not requiring that it extends it, merely that if you don't specify the `TError` then it defaults to be of type `Error`.
ie. this would be totally valid:
```TS
let info: ActionCompleted<LoggedIn, string> = ...
```

Reply:
@markwhitfeld I think good case if we usage implements Error, because we understand in advance that this is a type with an error
`export interface ActionCompletion<T = any, E extends Error = Error> {`